### PR TITLE
Add boundary condition to create a PeriodicMapAxis

### PR DESCRIPTION
**Pull request by AtreyeeS**
This PR adds a PeriodicMapAxis.
Its rather simple, only one function, so arguably it does not need its own class at all.
But I think its conceptually simpler to have a separate name for such an axis.
It can be useful for phase analysis, or for asymmetric IRFs defined not in (Lon, lat) but (offset, theta), as brought up in one of the CTAC ASWG calls...

### CODE REVIEW

**Review by MRegeard — APPROVED** [review]
Thanks @AtreyeeS, this is a great feature

**Comment by MRegeard** [comment]
On top of that, an idea that we could add following this feature is a method on maps to offset the data that are indexed through a periodic axis. This is usual in pulsar analysis to put the peak of the pulsar that would be around 0 (or 1) at the center of the phasogram.

File: gammapy/maps/tests/test_axes.py [hunk 1]
```diff
@@ -962,3 +969,15 @@ def test_energy_bin_per_decade_not_strict_bounds():
     )
 
     assert_allclose(axis.edges[0:-nbin] * 10.0, axis.edges[nbin:], rtol=1e-5, atol=0)
+
+
+def test_phase_map_axis():
+    axis1 = PeriodicMapAxis.from_bounds(-0.5, 0.5, 5)
+    coords = np.array([-1.0, -0.7, 0.1, 1.2])
+    assert_allclose(axis1.wrap_coords(coords), [0.0, 0.3, 0.1, 0.2], rtol=1e-5)
```
Comment (bkhelifi) [inline gammapy/maps/tests/test_axes.py:977]: A naive question.. the nodes of the final axis are not crescent?

**Comment by AtreyeeS** [comment]
I created an option on `MapAxis` as `boundary_type` to specify the boundary condition.
This still assumes that the edges wrap, or should that be configurable?

File: gammapy/maps/axes.py [hunk 2]
```diff
@@ -803,7 +836,7 @@ def pix_to_idx(self, pix, clip=False):
 
         return idx
 
-    def coord_to_pix(self, coord):
+    def _coord_to_pix(self, coord):
```
Comment (QRemy) [inline gammapy/maps/axes.py:839]: I don't think that the private definition is necessary here

File: gammapy/maps/axes.py [hunk 3]
```diff
@@ -778,6 +799,18 @@ def pix_to_coord(self, pix):
         values = self._transform.pix_to_coord(pix=pix)
         return u.Quantity(values, unit=self.unit, copy=False)
 
+    def _wrap_coords(self, coords):
```
Comment (QRemy) [inline gammapy/maps/axes.py:802]: could be public

File: gammapy/maps/axes.py [hunk 4]
```diff
@@ -820,7 +853,7 @@ def coord_to_pix(self, coord):
         pix = self._transform.coord_to_pix(coord=coord)
         return np.array(pix + self._pix_offset, ndmin=1)
 
-    def coord_to_idx(self, coord, clip=False):
+    def _coord_to_idx(self, coord, clip=False):
```
Comment (QRemy) [inline gammapy/maps/axes.py:856]: same

**Review by QRemy — COMMENTED** [review]
Thanks @AtreyeeS,  looks good, I made some change to remove the private functions.

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks @AtreyeeS and @QRemy! I have left some minor comment to clean up the implementation...

**Comment by QRemy** [comment]
Thanks @adonath for the suggestions I implemented them and I will merge this once the CI pass if no objection.

File: gammapy/maps/axes.py [hunk 5]
```diff
@@ -122,11 +141,18 @@ def __init__(self, nodes, interp="lin", name="", node_type="edges", unit=""):
         else:
             nodes = np.array(nodes)
 
+        available_boundary = {"periodic", "monotonic"}
+        if boundary_type not in available_boundary:
```
[the remark this answers is not in the document]
Comment (QRemy) [reply gammapy/maps/axes.py]: added
